fluidBuild.config.cjs

```diff
@@ -210,6 +211,7 @@ module.exports = {
 				"common/build/eslint-config-fluid/.*",
 				"common/lib/common-utils/jest-puppeteer.config.js",
 				"common/lib/common-utils/jest.config.js",
+        "common/build/eslint-plugin-fluid/.*",
```

Comment:
Indentation. But actually, can you check if things work if we make the rules `.cjs` files instead of plain `.js`? There's been a push towards being explicit in the repo when writing in plain JavaScript if a `.js` file is supposed to be CJS or ESM, instead of letting "the environment" (i.e. the package and its package.json settings) try to infer it.

Reply:
One slight problem with changing to `.cjs` is that the test suites aren't working as expected.

<img width="1327" alt="Screenshot 2024-02-15 at 15 19 03" src="https://github.com/microsoft/FluidFramework/assets/111468570/e2796f7a-fc17-4747-a2b1-ea9cadea472e">

To test the changes, I only applied `.cjs` change to one of the rules (`no-member-release-tag`), and the tests would only fail for the one with the `.cjs` extension.

+

Aside from the feedback, I also decided to move `rules` into the `src` directory.

Reply:
Tried `"test": "mocha --extension cjs src/test"` but this isn't also working. Let me look for more 😄 .

Reply:
Maybe also try with `.mjs`. I'm not an expert in this space but maybe the way they're written, they're ESM files, not CJS ones.